cpp/oneapi/dal/algo/decision_forest/detail/train_ops.hpp

```diff
@@ -38,28 +38,28 @@ struct train_ops {
     using descriptor_base_t = descriptor_base<task_t>;
 
     void check_preconditions(const Descriptor& params, const input_t& input) const {
+        using msg = dal::detail::error_messages;
+
         if (!(input.get_data().has_data())) {
-            throw domain_error("Input data should not be empty");
+            throw domain_error(msg::input_data_is_empty());
         }
         if (!(input.get_labels().has_data())) {
-            throw domain_error("Input labels should not be empty");
+            throw domain_error(msg::input_labels_are_empty());
         }
         if (input.get_data().get_row_count() != input.get_labels().get_row_count()) {
-            throw invalid_argument("Input data row_count should be equal to labels row_count");
+            throw invalid_argument(msg::input_data_rc_neq_input_labels_rc());
         }
```

Comment:
shouldn't we check if labels have the only column?

Reply:
We should, @makart19 need to ass this check to decision forest